python/lsst/consdb/hinfo_fix_missing.py

```diff
@@ -0,0 +1,258 @@
+import re
+from math import radians, sin
+from typing import Any
+
+import sqlalchemy as sa
+from astropy.coordinates import AltAz, EarthLocation, SkyCoord
+from astropy.time import Time
+from sqlalchemy.orm import Session
+
+from .utils import setup_logging, setup_postgres
+
+
+class Fixer:
+    """A collection of functions that try to patch up missing data columns."""
+
+    def __init__(self, exposure_rec: dict[str, Any], logger=None):
+        """Calls all fixup methods.
+
+        Given an `exposure_rec` dictionary, apply all fixup methods to the
+        dictionary in one go. Call this to apply the fixup methods. To add
+        more fixup methods, just add them to this class, make sure the
+        method name starts with "fix_", and make sure the signature matches:
+            def fix_whatever(self, exposure_rec: dict[str, Any])
+        The method should return a dictionary with keys to update.
+
+        """
+        self.logger = logger if logger is not None else setup_logging("hinfo_fix_missing")
+        self.updates: dict[str, Any]
+        self.updates = {}
+        for attr_name in dir(self):
+            if attr_name.startswith("fix_"):
+                attr = getattr(self, attr_name)
+                self.updates = {**self.updates, **attr(exposure_rec)}
+
+    def fix_telescope_position(self, exposure_rec: dict[str, Any]) -> dict[str, Any]:
+        """Attempts to fill in missing sky position columns.
+
+        It relies on s_ra and s_dec being present, and does
+        calculations to obtain altitude, azimuth and zenith
+        distance for the beginning, middle, and end of the
+        exposure as well as airmass.
+
+        It modifies the `exposure_rec` dictionary and returns nothing.
+
+        Parameters
+        ----------
+        exposure_rec : `dict[str, Any]`
+            The exposure record dictionary, (almost) ready to
+            be copied into consolidated database exposure
+            table.
+
+        Returns
+        -------
+        dict[str, Any]
+            Fixed columns in the original `exposure_rec` dictionary.
+        """
+
+        if exposure_rec["airmass"] is not None:
+            # No need to do calculations, it's already done.
+            return dict()
+
+        if exposure_rec["s_ra"] is None or exposure_rec["s_dec"] is None:
+            # Bail out because we don't have enough info.
+            return dict()
+
+        if exposure_rec["s_ra"] == 0.0 and exposure_rec["s_dec"] == 0.0:
```

Comment:
Did I see conversation about this somewhere on why we have these? could that info be added to the comment? Maybe I hallucinated it though

Reply:
I recall mentioning that there were zeroed out RA/Dec present in the database, but not how they got there. Really these values ought to be replaced with NULLs IMO.